### ISSUE TITLE
Request - transition support for style properties

### ISSUE THREAD

**Issue by reubenfirmin** [rjaros/kvision#259]
I have this very rough implementation.

```
    data class Transition(val property: String, val time: Double, val timingFunction: String? = null, val delay: Double? = null)

    fun Style.transition(transitions: List<Transition>) {
        setStyle("transition",
            transitions.joinToString(",") { trans ->
                "${trans.property} ${trans.time}s ${trans.timingFunction ?: ""} ${trans.delay ?: ""}"
            })
    }
```

...and I use like so:

```
    val decoratedCardStyleHover = Style(pClass = PClass.HOVER) {
        fontSize = 14 to px     
        marginLeft = 8 to px
        transition(listOf(Transition("margin-left", 0.1, "linear"),
                                 Transition("font-size", 0.1, "linear")))
    }
```

It would be cleaner if I could do something like:

```
    val decoratedCardStyleHover = Style(pClass = PClass.HOVER) {
        fontSize = 14 to px, withTransition(0.1, Easing.linear)     
        marginLeft = 8 to px, withTransition(0.1, Easing.easeOutSine)
        color = Color.name(Col.WHITE)
    }
```

**Comment by rjaros** [comment]
Unfortunately such DSL is not possible, because there is no way to automatically connect the property with the transition definition.
But I've added support for the new `transition` (and `transitionList`) properties for all KVision components and `Style` objects with a `Transition` class similar to yours.

**Comment by rjaros** [comment]
Fixed in 4.5.0
